Documentation/data-cleanup/resource_delete.py

```
@@ -0,0 +1,219 @@
+    return rows[0]["exists"]
+
+
+def drop_if_exists(table_name, sol_id):
+    """Drop table if exists and delete solution + metadata."""
+    logger.info(f"Checking table: {table_name}")
+
+    if table_exists(table_name):
+        logger.info(f"Dropping table: {table_name}")
+        db_execute(f'DROP TABLE IF EXISTS public."{table_name}" CASCADE;')
+
+        db_execute(f"DELETE FROM {ENV}_solutions WHERE solution_id = %s", (sol_id,))
```

Ruff (0.14.5)

Possible SQL injection vector through string-based query construction

(S608)